File: dev-docs/snippets/documents/open.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Open a document for editing. Returns the Yjs doc and its local metadata.
export async function open(client: JsBaoClient, documentId: string) {
  // #region example
  const { doc, metadata } = await client.documents.open(documentId, {
    waitForLoad: "localIfAvailableElseNetwork",
  });
  // #endregion example
  return { doc, metadata };
}

File: dev-docs/snippets/documents/open.swift
import JsBaoClient

// Open a document for editing. Swift returns an `OpenDocumentResult`
// (`doc`, `metadata`) and takes a typed `OpenDocumentOptions`.
func open(client: JsBaoClient, documentId: String) async throws {
  // #region example
  let result = try await client.documents.open(
    documentId,
    options: OpenDocumentOptions(waitForLoad: .localIfAvailableElseNetwork)
  )
  let doc = result.doc
  // #endregion example
  _ = doc
}
